service/storage_service.cc

```diff
@@ -5183,7 +5183,12 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
         auto guard = co_await _group0->client().start_operation(&_abort_source);
 
         while (_topology_state_machine._topology.is_busy()) {
-            rtlogger.debug("move_tablet(): topology state machine is busy");
+            using enum topology::transition_state;
+            const auto tstate = *_topology_state_machine._topology.tstate;
+            if (tstate == tablet_migration || tstate == tablet_migration) {
```

Comment:
tablet_migration repeated twice.

Also, I think qualifying with topology::transition_state would be better since it's clearer that they come from outer scope. I usually grep for at least the enum class when searching for usage: transition_state::tablet_migration.

Reply:
ack. fixed and now using fully qualified enum.